tests/Silverback.Integration.Tests.E2E/Kafka/MessageValidationTests.cs

```diff
@@ -0,0 +1,300 @@
+            DefaultTopic.MessagesCount.Should().Be(1);
+        }
+
+        [Fact]
+        public async Task Validation_WithWarning_InvalidMessageIsProduced()
+        {
+            var message = new TestValidationMessage
+            {
+                String10 = "1234567890abcd"
+            };
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedKafka())
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint
+                                        .ValidateMessage(false)
+                                        .ProduceTo(DefaultTopicName)))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+
+            Func<Task> act = () => publisher.PublishAsync(message);
+
+            await act.Should().NotThrowAsync<ValidationException>();
+            DefaultTopic.MessagesCount.Should().Be(1);
+        }
+
+        [Fact]
+        public async Task Validation_ThrowException_InvalidMessageNotConsumed()
+        {
+            bool received = false;
+
+            Host.ConfigureServices(
+                    services =>
+                    {
+                        services
+                            .AddLogging()
+                            .AddSilverback()
+                            .UseModel()
+                            .WithConnectionToMessageBroker(options => options.AddMockedKafka())
+                            .AddKafkaEndpoints(
+                                endpoints => endpoints
+                                    .Configure(
+                                        config =>
+                                        {
+                                            config.BootstrapServers = "PLAINTEXT://tests";
+                                        })
+                                    .AddOutbound<IIntegrationEvent>(
+                                        endpoint => endpoint.ProduceTo(DefaultTopicName))
+                                    .AddInbound(
+                                        endpoint => endpoint
+                                            .ConsumeFrom(DefaultTopicName)
+                                            .DeserializeJson(x => x.UseFixedType<TestValidationMessage>())
```

Comment:
We don't use `x` as a variable name over here. 😉

Reply:
Ouch sorry! 😏